```
std/digest/md.d(366): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._count)` assigned to longer lived variable `bits[0..8]`
std/digest/md.d(377): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[0])` assigned to longer lived variable `data[0..4]`
std/digest/md.d(378): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[1])` assigned to longer lived variable `data[4..8]`
std/digest/md.d(379): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[2])` assigned to longer lived variable `data[8..12]`
std/digest/md.d(380): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[3])` assigned to longer lived variable `data[12..16]`
std/digest/ripemd.d(532): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._count)` assigned to longer lived variable `bits[0..8]`
std/digest/ripemd.d(543): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[0])` assigned to longer lived variable `data[0..4]`
std/digest/ripemd.d(544): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[1])` assigned to longer lived variable `data[4..8]`
std/digest/ripemd.d(545): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[2])` assigned to longer lived variable `data[8..12]`
std/digest/ripemd.d(546): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[3])` assigned to longer lived variable `data[12..16]`
std/digest/ripemd.d(547): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[4])` assigned to longer lived variable `data[16..20]`
std/digest/md.d(366): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._count)` assigned to longer lived variable `bits[0..8]`
std/digest/md.d(377): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[0])` assigned to longer lived variable `data[0..4]`
std/digest/md.d(378): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[1])` assigned to longer lived variable `data[4..8]`
std/digest/md.d(379): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[2])` assigned to longer lived variable `data[8..12]`
std/digest/md.d(380): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[3])` assigned to longer lived variable `data[12..16]`
std/digest/ripemd.d(532): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._count)` assigned to longer lived variable `bits[0..8]`
std/digest/ripemd.d(543): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[0])` assigned to longer lived variable `data[0..4]`
std/digest/ripemd.d(544): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[1])` assigned to longer lived variable `data[4..8]`
std/digest/ripemd.d(545): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[2])` assigned to longer lived variable `data[8..12]`
std/digest/ripemd.d(546): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[3])` assigned to longer lived variable `data[12..16]`
std/digest/ripemd.d(547): Deprecation: slice of static array temporary returned by `nativeToLittleEndian(this._state[4])` assigned to longer lived variable `data[16..20]`
posix.mak:348: recipe for target 'generated/linux/debug/64/libphobos2.so.0.100.1' failed
```

--- a/compiler/src/dmd/escape.d
+++ b/compiler/src/dmd/escape.d
@@ -553,7 +553,16 @@ bool checkAssignEscape(Scope* sc, Expression e, bool gag, bool byRef)
         return false;
 
     if (e1.isSliceExp())
-        return false;
+    {
+        if (VarDeclaration va = expToVariable(e1))
+        {
+            if (!va.type.toBasetype().isTypeSArray() || // treat static array slice same as a variable
+                !va.type.hasPointers())
+                return false;
+        }
+        else
+            return false;
+    }
 
     /* The struct literal case can arise from the S(e2) constructor call:
      *    return S(e2);

--- a/compiler/test/fail_compilation/test17764.d
+++ b/compiler/test/fail_compilation/test17764.d
@@ -1,0 +1,21 @@
+/* REQUIRED_ARGS: -preview=dip1000
+ * TEST_OUTPUT:
+---
+fail_compilation/test17764.d(109): Error: scope variable `c` assigned to non-scope `global`
+---
+ */
+
+// https://issues.dlang.org/show_bug.cgi?id=17764
+
+#line 100
+
+int** global;
+
+struct S { int** str; }
+
+void f() @safe
+{
+    int* buf;
+    S[1] c = S(&buf);
+    global = c[0].str; /* This should be rejected. */
+}

